daemon/daemon.go

```diff
@@ -81,6 +81,16 @@ func (d *Daemon) Run() error {
 	writeTicker := time.NewTicker(d.config.WriteInterval)
 	defer writeTicker.Stop()
 
+	var labelTicker *time.Ticker
+	if d.config.LabelRefreshInterval > 0 {
+		labelTicker = time.NewTicker(d.config.LabelRefreshInterval)
+		defer labelTicker.Stop()
+	} else {
+		// Create dummy stopped ticker if label interval is not configured
+		labelTicker = time.NewTicker(time.Duration(1) * time.Hour)
+		labelTicker.Stop()
```

Comment:
Do we need a `defer` here too? Otherwise calling `Run` multiple times could lead to a timer leak? :sweat_smile: 

```suggestion
		defer labelTicker.Stop()
```

Reply:
Haha you're the Go-expert so please be patient with me ;)  : 

As I understand defer would only execute once the surrounding function call returns, is that correct?
In this particular case we want this dummy ticker to immediately stop so it won't ever run.
If we did defer labelTicker.stop() , wouldn't that mean that the ticker would run within the "time.Duration(1) * time.Hour" period because the stop call is only evaluated one Run() returns?

AFAIK the dummy ticker is only necessary so that the labelTicker.C select case won't break in cases that the refresh interval is set to 0, so maybe there is even a better solution to this (?).

Reply:
@miyunari  This is out of date now but I'd still like to know how this would behave so if you got a minute let me know.

Reply:
> Haha you're the Go-expert so please be patient with me ;) :

Please don't call me that :sweat_smile: 

But you are right. The `defer` would only call the function once `Run` returns. Looking at the patch itself, it did make little sense to me passing a specific time (`time.Hour`) into the timer. And stopping it immediately.

I made another suggestion that follows the logic https://github.com/RedHatInsights/host-metering/pull/27#discussion_r1400981591 :) Maybe it makes more sense :smile: